### DISCUSSION TITLE
Module parse failed: Bad character escape sequence (2:60) after v13.2.5-canary.20

### DISCUSSION THREAD

**Discussion by ayhanap** [vercel/next.js#47736] [Help]
### Summary

I am getting compile time error `Module parse failed: Bad character escape sequence (2:60)` after upgrading `v13.2.5-canary.20` and up. Tried to look at changelog but nothing seems to be related. Anyone else having similar issue?

Below is the console output..

### Additional information

```js
_ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from C:\Users\Ayhan\Documents\GitHub\userowl-app\.env.local
info  - Loaded env from C:\Users\Ayhan\Documents\GitHub\userowl-app\.env
warn  - You have enabled experimental feature (appDir) in next.config.js.
info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.

event - compiled client and server successfully in 1560 ms (268 modules)
wait  - compiling...
event - compiled client and server successfully in 134 ms (268 modules)
wait  - compiling /_error (client and server)...
event - compiled client and server successfully in 43 ms (269 modules)
warn  - Fast Refresh had to perform a full reload due to a runtime error.
wait  - compiling /(app)/widget/style/page (client and server)...
error - ./app/(app)/rootLayout.tsx
Module parse failed: Bad character escape sequence (2:60)
| import { createProxy } from "next/dist/build/webpack/loaders/next-flight-loader/module-proxy"
> const proxy = createProxy("C:\Users\Ayhan\Documents\GitHub\userowl-app\app\(app)\rootLayout.tsx")
|
| // Accessing the __esModule property and exporting $$typeof are required here.

Import trace for requested module:
./app/(app)/rootLayout.tsx
./app/(app)/layout.tsx
wait  - compiling...
wait  - compiling /(widget)/widget/[widgetSlug]/page (client and server)...
error - ./app/(app)/rootLayout.tsx
Module parse failed: Bad character escape sequence (2:60)
| import { createProxy } from "next/dist/build/webpack/loaders/next-flight-loader/module-proxy"
> const proxy = createProxy("C:\Users\Ayhan\Documents\GitHub\userowl-app\app\(app)\rootLayout.tsx")
|
| // Accessing the __esModule property and exporting $$typeof are required here.

Import trace for requested module:
./app/(app)/rootLayout.tsx
./app/(app)/layout.tsx
wait  - compiling /middleware (client and server)...
warn  - Fast Refresh had to perform a full reload due to a runtime error.
event - compiled successfully in 168 ms (62 modules)_
```

**Comment by icyJoseph** [comment]
Hi,

Could you share more details about your project? I just tested a fresh create-next-app with that canary version and it works fine.

Also try upgrading your canary version, there's next@13.2.5-canary.22 now.

**Reply by Xetoxyc** [reply]
We tried on .22

Seems to be a Windows problem. Our repo works well if i build with mac OS but i get the same escape error on my windows.

Could u validate this ?

Error came with next@13.2.5-canary.20

**Reply by ayhanap** [reply]
I also get this on windows 11, cannot validate on mac because my mac is on battery service for a couple of days.

**Reply by Xetoxyc** [reply]
Seems to be related to #47532  (packages/next/src/build/webpack/loaders/next-flight-loader/index.ts)
Also it seems like @feedthejim tried to revert this one with issue #47592

**Reply by xasanof17** [reply]
> Hi,
> 
> Could you share more details about your project? I just tested a fresh create-[next-app](https://linear.app/vercel/issue/NEXT-app) with that canary version and it works fine.
> 
> Also try upgrading your canary version, there's next@13.2.5-canary.22 now.

so how can we solve this problem bro. i have used sanity

**Comment by dingram94** [comment]
I am getting the same error with our project with next@13.2.5-canary.22

**Comment by ayhanap** [comment]
I tried reproducing it on a create-next-app, not the same error but I think it its definitely related. So I get this error on files marked with `'use client';`

My mac is on service so yes I get this error on Windows 11;

On next-create-app I get below error.

```
error - SyntaxError: "undefined" is not valid JSON
    at JSON.parse (<anonymous>)
```

Here is the reproducer. https://github.com/ayhanap/13.2.5-canary.22-reproducerr

**Reply by Xetoxyc** [reply]
True, seems to be related to "use client" as we also get errors only on "use client" components.

**Reply by hybrdthry911** [reply]
Same here. Updated to 13.3 everything broke. Tried a fresh install and it broke immediately upon seeing  'use client'

**Reply by Raveenrathnaraj** [reply]
Same here, tried to use client component inside the root layout, getting the same error. But client component inside the page file of layout is working.

**Comment by SuttonJack** [comment]
Also encountering this after upgrading to `13.2.5-canary.20` on Windows:

![image (2)](https://user-images.githubusercontent.com/92991945/229382749-9d259f1e-0350-4239-8f46-a7c7f62235a2.png)

Assuming it stems from https://github.com/vercel/next.js/pull/47532 - @shuding any idea why this PR would have broken `next dev` for Windows users?

**Reply by SuttonJack** [reply]
Some portion of the diff [here](https://github.com/vercel/next.js/pull/47532/files#diff-20be1066bb3785bdd9f255179957039e7ce035324bc7753aefe89ffc6bb6f1edL41-R75) may not be properly escaped? 

In terms of solutions, [This](https://www.reddit.com/r/node/comments/g4zdxe/how_to_fix_module_parse_failed_bad_character/) seems useful or
> **_All the components that has template url starting with letter "u" or if the template is under a folder whose name start with letter "u" results in the compilation error ""Bad character escape sequence" error on windows"_**

described [here](https://github.com/Urigo/angular-meteor/issues/1770)

**Reply by Xetoxyc** [reply]
there is already a "revert" from @feedthejim see #47592 feels like it gone to review but they could not revalidate the error jim got because they did not reviewed on a windows device. I already mentioned @feedthejim there to push this into further investigation

**Comment by iska9der** [comment]
Still catch this in 13.3.0

**Reply by SuttonJack** [reply]
Same - But this should at least make it hit their radar much faster. Lots of Windows users will presumably hit this

**Reply by SuttonJack** [reply]
13.3.1 fingers crossed

**Comment by hk86** [comment]
Same here, also on Windows starting with 13.2.5-canary.20:

```
error - SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
```

As mentioned by others this is related to `'use client'`.

**Reply by soymartinez** [reply]
full comparison: https://github.com/vercel/next.js/compare/v13.2.5-canary.19...v13.2.5-canary.20

**Comment by abdulra7man-shaban** [comment]
getting the same error after updating to nextjs 13.3

**Reply by tlgsn** [reply]
getting error after 13.3 update

**Comment by romulororiz** [comment]
getting the same after updating it

**Comment by Xetoxyc** [comment]
There is a PR https://github.com/vercel/next.js/issues/47704#issuecomment-1501058527 which should fix this issue. Windows user should wait for this PR to get merged and update to canary or downgrade afterwards

**Reply by incepter** [reply]
It is just sad to see that Microsoft is doing efforts to help developers all around the globe, and that developers will massively and heavily use its products for their own good, without supporting microsoft ecosystem! Windows specific issues continue to strike every time! Now we have to "Wait".

**Comment by miradontsoa** [comment]
It is solved finally in 13.3.1-canary.4

**Reply by bhojkamal** [reply]
Thanks. worked on 13.3.1-canary.4 onward.

**Comment by Caknoooo** [comment]
you can update next js to 13.3.1-canary.4 

npm
```
npm i next@13.3.1-canary.4
```

yarn 
```
yarn add next@13.3.1-canary.4
```

Just Work for me ^_^

**Reply by batyr-qhan** [reply]
great!

**Reply by eriktonon** [reply]
really helped a lot

**Comment by javiermaciasa** [comment]
I'm getting the same error after updating

**Reply by iska9der** [reply]
Follow [this](https://github.com/vercel/next.js/discussions/47736#discussioncomment-5576280) instruction, then remove `.next` folder from root of your project, and run `npm run build`

**Reply by javiermaciasa** [reply]
It worked thanks

**Comment by eosfelipe** [comment]
Solved by updating nextjs to 13.3.1-canary.4

**Reply by clxrityy** [reply]
Solved the issue for me as well!

**Reply by brixtonpham** [reply]
thank you it's worked for me

**Comment by Wence17** [comment]
I'm having the same error

**Comment by igmeMarcial** [comment]
![Captura de pantalla 2023-04-13 133931](https://user-images.githubusercontent.com/88986288/232249073-1124bf70-81f5-43d1-bdf8-e3296f8155d5.png)
i have this error ,I do not understand why

**Reply by TimothyCorreia-Paul** [reply]
This comment thread has your answer
https://github.com/vercel/next.js/discussions/47736#discussioncomment-5584219
